java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java

```diff
@@ -80,6 +83,40 @@
           tempGcsClient.getInternalConfig().entrySet()) {
         rayConfig.rayletConfigParameters.put(entry.getKey(), entry.getValue());
       }
+
+      // Keep this method logic in sync with `services.get_address_info_from_redis_helper`
+      int numRetries = 5;
+      int retryCount = 0;
+      boolean loadConfig = false;
+      while (retryCount++ < numRetries) {
+        for (NodeInfo nodeInfo : tempGcsClient.getAllNodeInfo()) {
+          if (rayConfig.nodeIp.equals(nodeInfo.nodeAddress) ||
+              (nodeInfo.nodeAddress.equals("127.0.0.1") &&
+                  rayConfig.nodeIp.equals(rayConfig.getRedisAddress()))) {
+            rayConfig.objectStoreSocketName = nodeInfo.objectStoreSocketName;
```

Comment:
This whole section should be put into the constructor instead of the static block. The `rayConfig` here is just a local variable. I don't understand how it works because the changes you made here will be lost.

Reply:
`RayConfig` is a singleton. We need to use `ray config` to load native binary and library to specified directory. If we do this in the `RayConfig` constrcutor, how can we get ray config instance?

Reply:
There's actually a `reset()` method in `RayConfig`, so `RayConfig` is not fully singleton. The global instance will be destroyed during the shutdown process. I was saying we should put this into the constructor of `RayNativeRuntime`, not of `RayConfig`.

Reply:
Fixed